packages/sitecore-jss-dev-tools/src/package-deploy.ts

```diff
@@ -163,25 +182,20 @@ async function watchJobStatus(options: PackageDeployOptions, taskName: string) {
   const factors = [options.appName, taskName, `${options.importServiceUrl}/status`];
   const mac = hmac(factors, options.secret);
 
-  const isHttps = options.importServiceUrl.startsWith('https');
-
   const requestBaseOptions = {
-    transport: isHttps ? getHttpsTransport(options) : undefined,
     headers: {
       'User-Agent': 'Sitecore/JSS-Import',
       'Cache-Control': 'no-cache',
       'X-JSS-Auth': mac,
     },
-    proxy: extractProxy(options.proxy),
-    maxRedirects: 0,
-    httpsAgent: isHttps
-      ? new HttpsAgent({
-          // we turn off normal CA cert validation when we are whitelisting a single cert thumbprint
-          rejectUnauthorized: options.acceptCertificate ? false : true,
-          // needed to allow whitelisting a cert thumbprint if a connection is reused
-          maxCachedSessions: options.acceptCertificate ? 0 : undefined,
-        })
-      : undefined,
+    dispatcher: new ProxyAgent({
+      uri: options.proxy || '',
+      maxRedirections: 0,
+      connect: {
+        rejectUnauthorized: options.acceptCertificate ? false : true,
```

Comment:
We need to leave comments (that you removed) in our code, to have understanding of why it was configured